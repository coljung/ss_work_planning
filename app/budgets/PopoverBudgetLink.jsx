import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import { Popover } from 'antd';

import { ROUTE_BUDGET } from '../Routes';

const viewsTop = [
    { name: 'Executive Recap', url: 'exec' },
    { name: 'OTB Total', url: 'total' },
    { name: 'OTB Women', url: 'women' },
    { name: 'OTB Men', url: 'men' },
];
const viewsMiddle = [
    { name: 'Brand Group Targets Women', url: 'bg-women' },
    { name: 'Brand Group Targets Men', url: 'bg-men' },
    { name: 'Department Targets Women', url: 'department' },
    { name: 'Department Targets Men', url: 'department2' },
    { name: 'Width and Depth Targets', url: 'width-depth' },
];
const viewsBottom = [
    { name: 'Delivery', url: 'delivery' },
    { name: 'Unit', url: 'unit' },
    { name: 'Department', url: 'department3' },
    { name: 'Department', url: 'department4' },
    { name: 'Sales Margin', url: 'sales-margin' },
    { name: 'Shrink Planning', url: 'shrink-planning' },
];

const PopoverbudgetLink = ({ budgetId, seasonName, versionId, versionName }) => {
    const text = <span>{seasonName} views</span>;

    const displayLinkFunction = link =>
        <li key={`${seasonName}-${link.index}-${link.url}`}>
            <Link id={`${seasonName}-${link.index}-${link.url}`}
                  to={`${ROUTE_BUDGET}/${seasonName}/budget/${budgetId}/version/${versionName}/${versionId}/${link.url}`}>
                {link.name}
            </Link>
        </li>;

    const top = viewsTop.map(displayLinkFunction);
    const middle = viewsMiddle.map(displayLinkFunction);
    const bottom = viewsBottom.map(displayLinkFunction);

    const content = (
      <ul className='popover-main-ul'>
          <li>
              <h4>Top Down</h4>
              <ul>{top}</ul>
          </li>
          <li>
              <h4>Middle Out Summary</h4>
              <ul>{middle}</ul>
          </li>
          <li>
              <h4>Bottom Up</h4>
              <ul>{bottom}</ul>
          </li>
      </ul>
    );

    return (
        <Popover placement="right" title={text} content={content} trigger="click">
            {seasonName}
        </Popover>
    );
};

PopoverbudgetLink.propTypes = {
    budgetId: PropTypes.string,
    seasonName: PropTypes.string,
    versionId: PropTypes.string,
    versionName: PropTypes.string,
};

export default PopoverbudgetLink;
