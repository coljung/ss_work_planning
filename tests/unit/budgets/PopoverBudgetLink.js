import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Button } from 'antd';
import PopoverBudgetLink from '../../../app/budgets/PopoverBudgetLink';

describe('PopoverBudgetLink', () => {
    it('should render correctly', () => {
        const output = shallow(
            <PopoverBudgetLink title="fsdfsdf">Hello Jest!</PopoverBudgetLink>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it.skip('should render correctly with btnInTitle', () => {
        const output = shallow(
            <Board title="Hello World!" btnInTitle={<Button>Foo Bar</Button>}>Hello Jest!</Board>
        );

        expect(output.find('Button').render().text()).toBe('Foo Bar');
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
