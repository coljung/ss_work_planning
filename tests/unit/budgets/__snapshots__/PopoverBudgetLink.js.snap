// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverBudgetLink should render correctly 1`] = `
<Popover
  content={
    <ul
      className="popover-main-ul"
    >
      <li>
        <h4>
          Top Down
        </h4>
        <ul>
          <li>
            <Link
              id="undefined-0-exec"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/exec"
            >
              Executive Recap
            </Link>
          </li>
          <li>
            <Link
              id="undefined-1-total"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/total"
            >
              OTB Total
            </Link>
          </li>
          <li>
            <Link
              id="undefined-2-women"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/women"
            >
              OTB Women
            </Link>
          </li>
          <li>
            <Link
              id="undefined-3-men"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/men"
            >
              OTB Men
            </Link>
          </li>
        </ul>
      </li>
      <li>
        <h4>
          Middle Out Summary
        </h4>
        <ul>
          <li>
            <Link
              id="undefined-0-bg-women"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/bg-women"
            >
              Brand Group Targets Women
            </Link>
          </li>
          <li>
            <Link
              id="undefined-1-bg-men"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/bg-men"
            >
              Brand Group Targets Men
            </Link>
          </li>
          <li>
            <Link
              id="undefined-2-department"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/department"
            >
              Department Targets Women
            </Link>
          </li>
          <li>
            <Link
              id="undefined-3-department2"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/department2"
            >
              Department Targets Men
            </Link>
          </li>
          <li>
            <Link
              id="undefined-4-width-depth"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/width-depth"
            >
              Width and Depth Targets
            </Link>
          </li>
        </ul>
      </li>
      <li>
        <h4>
          Bottom Up
        </h4>
        <ul>
          <li>
            <Link
              id="undefined-0-delivery"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/delivery"
            >
              Delivery
            </Link>
          </li>
          <li>
            <Link
              id="undefined-1-unit"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/unit"
            >
              Unit
            </Link>
          </li>
          <li>
            <Link
              id="undefined-2-department3"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/department3"
            >
              Department
            </Link>
          </li>
          <li>
            <Link
              id="undefined-3-department4"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/department4"
            >
              Department
            </Link>
          </li>
          <li>
            <Link
              id="undefined-4-sales-margin"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/sales-margin"
            >
              Sales Margin
            </Link>
          </li>
          <li>
            <Link
              id="undefined-5-shrink-planning"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/season/undefined/budget/undefined/version/undefined/undefined/shrink-planning"
            >
              Shrink Planning
            </Link>
          </li>
        </ul>
      </li>
    </ul>
  }
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  overlayStyle={Object {}}
  placement="right"
  prefixCls="ant-popover"
  title={
    <span>
       views
    </span>
  }
  transitionName="zoom-big"
  trigger="click"
/>
`;
